daliuge-engine/dlg/parset_drop.py

```diff
@@ -82,7 +82,7 @@ def initialize(self, **kwargs):
         self.config_data = self.serialize_parameters(
             self.filter_parameters(self.parameters, self.mode), self.mode).encode('utf-8')
 
-    def getIO(self):
+    def _getIO(self):
```

Comment:
A rather large refactor, but I haven't found an existing case where the DataDROP interface isn't sufficient for an app and needs to use DataIO directly. For streaming I've simply added dataDrop.writeStream() and dataDrop.readStream()